### PR TITLE
Topic page redesign

### LINKED ISSUE #154: Get rid of Tags

### PR DESCRIPTION
**Pull request by jszymanowski**
Closes #154, #159, #158, #156, #163, #150

### CODE REVIEW

**Comment by ARRiera96** [comment]
@omriBernstein Check it outtt!

File: .gitignore [hunk 1]
```diff
@@ -1,7 +1,8 @@
 .idea
 node_modules
 npm-debug.log
-public
```
Comment (omriAckley) [inline .gitignore:4]: This should have covered it—not that your change is problematic, just wondering what triggered the change.

File: browser/js/topics/topic.html [hunk 2]
```diff
@@ -9,12 +9,12 @@
           <!-- prerequisite topic title -->
           <a ui-sref="topic({ topicId: prereq.prerequisiteId })">{{ prereq.title }}</a>
```
Comment (omriAckley) [inline browser/js/topics/topic.html:10]: This file is pretty large. Consider refactoring by porting some of the HTML here into various directives.

File: browser/js/topics/topic.html [hunk 3]
```diff
@@ -46,83 +53,80 @@
...
+                      <span class="btn btn-xs btn-default">
+                        <span class="glyphicon glyphicon-pencil" ng-if="resource.type === 'tutorial'"></span>
+                        <span class="glyphicon glyphicon-film" ng-if="resource.type === 'video'"></span>
+                        <span class="glyphicon glyphicon-text-height" ng-if="resource.type === 'article'"></span>
+                        <span class="glyphicon glyphicon-book" ng-if="resource.type === 'book'"></span>
+                        <span class="glyphicon glyphicon-th-list" ng-if="resource.type === 'documentation'"></span>
+                        {{ resource.type }}
+                      </span>
+                      <!-- Voting -->
+                      <button id="btn-resource-upvote-{{ resource.id }}" type="button" class="btn btn-xs btn-default" ng-click="upvote('resource', resource.id)">
```
Comment (omriAckley) [inline browser/js/topics/topic.html:93]: Are these `id`s necessary? If you're using them for styling I'd recommend switching to a class—I think it leads to more reusable CSS practices. If you're using them in your JS then this should probably be a directive.

File: browser/js/topics/topic.js [hunk 4]
```diff
@@ -23,7 +23,7 @@ app.controller('TopicCtrl', function ($scope, TopicFactory, topic, VoteFactory,
```
Comment (omriAckley) [inline browser/js/topics/topic.js:23]: This file is also pretty large, I'd recommend splitting it up. Large files can 1) make code hard to find, and 2) lead to unnecessary merge conflicts when two people are working on two independent features but happen to touch the same file.

**Comment by omriAckley** [comment]
Keep up the good work! Probably my biggest piece of advice looking at this code slice is watch out for huge files—I'd recommend splitting them up. They're too big to not fail.

File: .gitignore (hunk 1 above)
Comment (jszymanowski) [inline .gitignore:4]: We added an images folder to /public for our logo, which doesn't really need to be part of the repository, but seemed the easiest way to sync amongst ourselves.

File: browser/js/topics/topic.html (hunk 3 above)
Comment (jszymanowski) [inline browser/js/topics/topic.html:93]: They're used in a jQuery fn to show/hide the upvote/voted button.  We'll look to shift the resource boxes to a directive.

**Pull request merged**